### DISCUSSION TITLE
need help with subtitle JS template

### DISCUSSION THREAD

**Discussion by Mariusthvdb** [elchininet/custom-sidebar#197] [Q&A]
testing the new functionality for subtitle (thanks!) I need to replace what I use in card_mod:

```
      .menu .title:after {
        /*content: "{%- set m_alerts = states('sensor.marquee_alerts')|int(0) %}
                  {%- set phrase = 'Alert: ' if m_alerts == 1 else 'Alerts: ' -%}
                  {{phrase ~ m_alerts if m_alerts > 0 }}" !important;
        white-space: pre;
        font-size: 12px;*/
        font-weight: bold;
        /*display: block;
        color: var(--alert-color) !important;*/
        animation: marquee 4s alternate infinite;
      }
```

as you can see Ive already taken out most of that, except for the styling (bold, and animation)

pasting the exact Jinja template works perfectly, so that proves things are working correctly, even the combined settings with card_mod.

now Id like to change the jinja template to JS, and use the partials.

```
js_variables:
  marquee: sensor.marquee_alerts

partials:
  subtitle: |
    const marquee = states(marquee);
```

are accepted (no errors on the log)

But I can not get it to work in the actual config with:

```
subtitle: >
  [[[ @partial subtitle;
      return 'Alerts: ' + marquee; ]]]
```

not even a hardcoded:
```
subtitle: >
  [[[ @partial subtitle;
      return 'Alerts: 3'; ]]]
```

works. I have to take out the @partial :

```
subtitle: >
  [[[ return 'Alerts: 3'; ]]]
```

to get it functional again.....

what am I doing wrong here?

**Comment by Mariusthvdb** [comment]
ignore me, I had used the same variable name twice.....


made it work with:
```
partials:
  subtitle: |
    const marquees = states(marquee);
    const phrase = marquees == 1 ? 'Alert: ' : 'Alerts: ';
```

and:

```
subtitle: >
  [[[ @partial subtitle;
      return phrase + marquees; ]]]
```

![Nov-12-2024 09-01-45](https://github.com/user-attachments/assets/8abd5cbd-ec80-43ea-8303-26f952dcb1c1)

**Comment by elchininet** [comment]
I am seeing that you already sorted it out. 👍🏼

Just something that I am seeing in your code. `@partial something` doesn't need a semicolon at the end.
